reword introduction for clarity (#294)
NixOS/nixfmt: README.md

# `nixfmt`

`nixfmt` is the official formatter for Nix language code, intended to easily apply a uniform style.

![Build Status](https://github.com/NixOS/nixfmt/actions/workflows/main.yml/badge.svg?branch=master)

## State

`nixfmt` was originally developed by [Serokell](https://github.com/serokell).

> It was used as the basis for the official standardised Nix formatter, as established by [RFC 166](https://github.com/NixOS/rfcs/pull/166).

Reviewer: This feels slightly off. It reads as if the current project _was_ used as the basis of some other project that came out of rfc166.

I believe this could be clarified:
```suggestion
The original `nixfmt` formed the basis of [RFC 166](https://github.com/NixOS/rfcs/pull/166), which this project now implements.
```

or maybe:

```suggestion
The original `nixfmt` formed the basis of [RFC 166](https://github.com/NixOS/rfcs/pull/166), which this project will implement in the [upcoming release](https://github.com/NixOS/nixfmt/issues/272).
```

The official standard differs considerably from the original implementation.

> Be aware of this if you track the [`master`](https://github.com/NixOS/nixfmt/tree/master) branch.

Reviewer: This sentence is ambiguous; it could either mean "be aware the `master` branch is outdated, following the old style, while development happens on another branch" **or** it could mean the opposite: "be aware the `master` branch is unstable, making frequent breaking changes as we work towards the upcoming stable release implementing the new official standard".

I believe the latter is the correct interpretation, but I'm unsure how to phrase it clearly. Perhaps something like:

```suggestion
Be aware the [`master`](https://github.com/NixOS/nixfmt/tree/master) branch is now implementing the new official standard, **not** the original style used by the [latest release](https://github.com/NixOS/nixfmt/releases/tag/v0.6.0).
```

Author: ```suggestion
Be aware the [`master`](https://github.com/NixOS/nixfmt/tree/master) branch implements the new official standard, **not** the original style from the [last release made by Serokell](https://github.com/NixOS/nixfmt/releases/tag/v0.6.0).
```

But that repeats half of the other sentences, so I'd add that bit in coordination with the other statements.

> Until the first new release, `nixfmt` should be considered **very unstable**.

Reviewer 2: (not your fault, feel free to ignore):

I don't think it would be clear to readers what "the first new release" means here. Should we link to https://github.com/NixOS/nixfmt/issues/272 instead?

> Until the [next release](https://github.com/NixOS/nixfmt/issues/272), `nixfmt` should be considered **very unstable**.

Reviewer: Slight wording tweak:
```suggestion
Until a new [stable release](https://github.com/NixOS/nixfmt/issues/272) is made, `nixfmt` should be considered **very unstable**.
```

Author: Yeah but "new" can also be read as "latest", i.e. already there, while "next" is clearly about the future. This should be possible to infer without reading the linked issue. Also in this context there's no way to know what makes a release "stable".

Author: ```suggestion
Until the [next release](https://github.com/NixOS/nixfmt/issues/272), expect `nixfmt` to be subject to change.
```

How about this?

> A recent version of `nixfmt` is available as `pkgs.nixfmt-rfc-style` in Nixpkgs.

Reviewer: Do we want to be consistent with the "upcoming release" distinction here?

```suggestion
A recent snapshot of `nixfmt`'s upcoming release is available as `pkgs.nixfmt-rfc-style` in Nixpkgs.
```

Reviewer: > What's a snapshot of a release?

Good point.

I guess I was _trying_ to suggest clarifying that this relates to an as-yet unreleased version.

> The original `nixfmt` is also still available as `pkgs.nixfmt-classic` for now, though unmaintained.

Author: Parenthesis tend to break the reading flow, and that particular way of inserting a statement can be laid out into a proper sentence.

```suggestion
The original `nixfmt` is also still available as `pkgs.nixfmt-classic` for now, but it is unmaintained and will eventually be removed.
```
